Rise-Vision/rise-video
# Feature/responsive behavior

## Description
Add support for videos to fit the size of their template container.

## Motivation and Context
Users of the component will be able to control the size of the videos by wrapping them in an element with the desired size.

Implemented according to [this section of the component design](https://docs.google.com/document/d/1LG_h_jQX8UlGLHfmZhEch0DCQyeVxmCnnDAUyU0CaoQ/edit#heading=h.t9ql0a4zg0do).

Additionally, Mathew Meiers indicated in [this Trello card](https://trello.com/c/RmQBDmxZ) that it would be helpful if the "black bars"  which appear when videos are smaller than their container could be made transparent, and it was trivial to do so, so that functionality was added.

I initially attempted to use the fluid feature of `videojs`, but realized that this feature doesn't work as intended when inside a shadow dom, as it relies on injecting stylesheets into the `<head>` element.

I realized that the native `<video>` element already handles scaling and centering videos so all that was needed was to scale the `<video>` element to the size of the parent container of the component and let it do the heavy lifting.

## How Has This Been Tested?
- Integration tests have been added for new code. These are minimal, since most of the heavy lifting is done by the HTML5 `<video>` element.
- Changes have been tested on both Electron and Chrome extension players using [this presentation](https://apps.risevision.com/templates/edit/7900dcc5-72c0-46e1-b67f-29d96bc578a0/?cid=7fa5ee92-7deb-450b-a8d5-e5ed648c575f) which uses a test template with a number of videos wrapped in differently sized containers. The container element for each video has a green background to demonstrate transparency.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
  - Manual Test: Completed as noted above
  - Automated Test: Completed as noted above
  - Monitoring: All requirements met
  - Rollback: The previous release has been tagged so rollback should be simple
  - Documentation: The documentation has been updated with information about new responsive behaviour and how "black bars" and transparency are handled
  - Internal Communication: Support does not need to be notified as the component is not in use
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
- No release checklist items were skipped
